sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/MigrationGuideSnippetsLiveTests.cs

```diff
@@ -0,0 +1,192 @@
+            var blobContainerName = "<< NAME OF THE BLOB CONTAINER  >>";
+            var legacyBlobContainerName = "<< NAME OF THE BLOB CONTAINER THAT CONTAINS THE LEGACY DATA>>";
+            /*@@*/
+            /*@@*/ storageConnectionString = StorageTestEnvironment.Instance.StorageConnectionString;
+            /*@@*/ blobContainerName = storageScope.ContainerName;
+
+            using var cancellationSource = new CancellationTokenSource();
+
+            // Read the legacy checkpoints; these are read eagerly, as the number of partitions
+            // for an Event Hub is limited so the set should be a manageable size to hold in memory.
+            //
+            // Note: The ReadLegacyCheckpoints method will be defined in another snippet.
+
+            /*@@*/ var legacyCheckpoints = ReadFakeLegacyCheckpoints(legacyBlobContainerName);
+            /*@@*/
+            //@@var legacyCheckpoints = await ReadLegacyCheckpoints(
+                //@@storageConnectionString,
+                //@@legacyBlobContainerName,
+                //@@consumerGroup,
+                //@@cancellationSource.Token);
+
+            // The member names of MigrationCheckpoint match the names of the checkpoint
+            // names of the checkpoint metadata keys.
+
+            var offsetKey = nameof(MigrationCheckpoint.Offset).ToLowerInvariant();
+            var sequenceKey = nameof(MigrationCheckpoint.SequenceNumber).ToLowerInvariant();
+
+            // The checkpoint blobs require a specific naming scheme to be valid for use
+            // with the EventProcessorClient.
+
+            var prefix = string.Format(
+                "{0}/{1}/{2}/checkpoint/",
+                fullyQualifiedNamespace.ToLowerInvariant(),
+                eventHubName.ToLowerInvariant(),
+                consumerGroup.ToLowerInvariant());
+
+            // Create the storage client to write the migrated checkpoints.  This example
+            // assumes that the connection string grants the appropriate permissions to create a
+            // container in the storage account.
+
+            var storageClient = new BlobContainerClient(storageConnectionString, blobContainerName);
+            await storageClient.CreateIfNotExistsAsync(cancellationToken: cancellationSource.Token);
+
+            // Translate each of the legacy checkpoints, storing the offset and
+            // sequence data into the correct blob for use with the EventProcesorClient.
+
+            foreach (var checkpoint in legacyCheckpoints)
+            {
+                var metadata = new Dictionary<string, string>()
+                {
+                    { offsetKey, checkpoint.Offset.ToString(CultureInfo.InvariantCulture) },
+                    { sequenceKey, checkpoint.SequenceNumber.ToString(CultureInfo.InvariantCulture) }
+                };
+
+                BlobClient blobClient = storageClient.GetBlobClient($"{ prefix }{ checkpoint.PartitionId }");
```

Comment:
Surprised that you used BlobClient instead of var. Lol. :D

Reply:
I try to follow the "explicit type when not a constructor" approach for these, since there's no easy way to hover/nav to the definition.... but it feels so wrong writing it out.  :smile: